### PR TITLE
Update autoconf, krb5, m4 and readline

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update the following packages to the latest versions: autoconf, krb5 (upgrade needed to compile with latest autoconf), m4 and readline

###### Change Log  

- Update specs for autoconf, krb5, m4 and realine
- Update cgmanifest
- Update temp toolchain scripts to compile latest versions

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/autoconf/autoconf.spec [hunk 1]
```diff
@@ -1,14 +1,13 @@
 Summary:	    The package automatically configure source code
 Name:		    autoconf
-Version:	    2.69
-Release:        11%{?dist}
+Version:	    2.71
+Release:        1%{?dist}
 License:	    GPLv2
-URL:		    http://www.gnu.org/software/autoconf
+URL:		    https://www.gnu.org/software/autoconf/
```
Comment (PawelWMS) [inline SPECS/autoconf/autoconf.spec:6]: Small request: could run a linter against the .spec files to get rid of tabs/fix alignment?

File: SPECS/krb5/krb5.spec [hunk 2]
```diff
@@ -41,11 +41,10 @@ These are the additional language files of krb5.
...
 
 %build
-cd src &&
-sed -e 's@\^u}@^u cols 300}@' \
-    -i tests/dejagnu/config/default.exp &&
-CPPFLAGS="-D_GNU_SOURCE %{getenv:CPPFLAGS}" \
-autoconf &&
+cd src
+sed -e 's@\^u}@^u cols 300}@' -i tests/dejagnu/config/default.exp
+#autoupdate
```
Comment (PawelWMS) [inline SPECS/krb5/krb5.spec:46]: What is this comment about? Could you either remove or extend it a bit?

**Review by PawelWMS — APPROVED** [review]

File: SPECS/krb5/krb5.spec (hunk 2 above)
Comment (anphel31) [reply SPECS/krb5/krb5.spec:46]: Just a test - will remove

File: SPECS/autoconf/autoconf.spec (hunk 1 above)
Comment (anphel31) [reply SPECS/autoconf/autoconf.spec:6]: Sure thing

**Review by jslobodzian — APPROVED** [review]

**Comment by anphel31** [comment]
Closing in favor of #1745 which targets the main branch

**Pull request closed without merging**